src/policy/protect_payload.rs

```diff
@@ -1,21 +1,38 @@
 //! The protect payload policy, it protects the payload.
 
+use core::slice;
+use core::sync::atomic::{AtomicBool, Ordering};
+
+use miralis_abi::failure;
 use miralis_core::abi_protect_payload;
+use tiny_keccak::{Hasher, Sha3};
 
 use crate::arch::pmp::pmpcfg;
 use crate::arch::pmp::pmplayout::POLICY_OFFSET;
 use crate::arch::MCause::EcallFromSMode;
 use crate::arch::{MCause, Register};
+use crate::config::PAYLOAD_HASH_SIZE;
 use crate::host::MiralisContext;
 use crate::policy::{PolicyHookResult, PolicyModule};
 use crate::virt::{RegisterContextGetter, VirtContext};
 
+const LINUX_LOCK_PAYLOAD_HASH: [u8; 32] = [
+    241, 90, 158, 184, 200, 210, 145, 178, 30, 80, 200, 161, 56, 120, 75, 241, 68, 38, 21, 2, 248,
+    112, 128, 155, 31, 240, 37, 94, 203, 66, 243, 167,
+];
+
+const TEST_POLICY_PAYLOAD: [u8; 32] = [
+    140, 165, 21, 156, 234, 234, 86, 210, 171, 224, 126, 232, 161, 252, 107, 57, 157, 200, 184,
+    100, 29, 28, 164, 226, 250, 171, 137, 218, 97, 135, 196, 86,
+];
```

Comment:
Keep in mind that we will need to check the hash of different kernels, for instance the one on the board is a issued from a fork maintained by StarFive, so we need support for at least two kernel images and the test payload.